pkg/reconcile/reconcilers.go

```diff
@@ -156,8 +157,14 @@ func (r *defaultReconciler) ReconcileSchema(ctx context.Context, req ReaperReque
 		return nil, nil
 	}
 
+	// Check that Cassandra is ready before creating the schema job if running under k8ssandra
+	err := r.checkForCassandraDatacenterReadiness(ctx, req)
```

Comment:
See my comment below about `checkForCassandraDatacenterReadiness`. Basically the check should not block and should fail fast.